### ISSUE TITLE
[FEATURE] Docker named events and/or signals command

### ISSUE THREAD

**Issue by shaunwarman** [docker/docker#31947]
I hope something similar doesn't exist, but it would be nice to have an explicitly named event trigger during certain phases of a docker container starting up.

Some command in a `Dockerfile` like:
```
...
EVENT ["custom-event"]
...
EVENT ["ready"]
...
```

and in `Docker-compose.yml` a way to control startup order via these events from healthcheck or start on like command.
```
...
healthcheck:
  event: "custom-event"
...
OR
...
starton:
  event: "custom-event"
...
```

**Comment by cpuguy83** [comment]
Docker already has events, I do not think there should be custom events.
What we don't seem to do is emit an event when a container's health state changes.

**Comment by shaunwarman** [comment]
Correct. I can listen to docker events from control socket (`docker.sock`), but how could that event get emitted or listened to easily from say a docker compose file. I am just thinking of how to more easily control startup order of dependent containers.

**Comment by cpuguy83** [comment]
Actually looking further, there is an event already called "health_status"
